src/main/java/com/suse/saltstack/netapi/event/BeaconEvent.java

```diff
@@ -0,0 +1,46 @@
+package com.suse.saltstack.netapi.event;
+
+import com.suse.saltstack.netapi.datatypes.Event;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class BeaconEvent {
+    private static final Pattern ret = Pattern.compile("^salt/beacon/([^/]+)/([^/]+)/(.+)$");
+
+    private final String beacon;
+    private final String minionId;
+    private final String path;
+    private final Map<String, Object> data;
+
+    public BeaconEvent(String minionId, String beacon, String path, Map<String, Object> data) {
+        this.minionId = minionId;
+        this.beacon = beacon;
+        this.path = path;
+        this.data = data;
+    }
+
+    public String getBeacon() {
+        return beacon;
+    }
+
+    public String getMinionId() {
+        return minionId;
+    }
+
+    public Map<String, Object> getData() {
+        return data;
+    }
+
+    public static Optional<BeaconEvent> parse(Event event) {
+        Matcher m = ret.matcher(event.getTag());
```

Comment:
What is an acceptable minimum? As an example, which one of following would be acceptable? (I guess option 1 is there for posterity)

    1. for (int i = 0; i < pos; ++i)
    2. for (int idx = 0; idx < pos; ++idx)
    3. for (int index = 0; index < position; ++index)